datadog/fwprovider/resource_datadog_csm_resource_evaluation_filter.go

```
@@ -0,0 +1,293 @@
+	out := make([]attr.Value, len(s))
+	for i, v := range s {
+		out[i] = types.StringValue(v)
+	}
+	return out
+}
+
+func (r *CsmResourceEvaluationFilter) UpdateState(_ context.Context, state *ResourceEvaluationFilterModel, attributes *datadogV2.ResourceFilterAttributes) {
+	for p, accounts := range attributes.CloudProvider {
+		for id, tagList := range accounts {
+			tags := types.ListValueMust(types.StringType, convertStringSliceToAttrValues(tagList))
+			state.CloudProvider = types.StringValue(p)
+			state.ID = types.StringValue(id)
+			state.Tags = tags
+			break
+		}
+		break
+	}
+}
+
+func (r *CsmResourceEvaluationFilter) Read(ctx context.Context, request resource.ReadRequest, response *resource.ReadResponse) {
+	var state ResourceEvaluationFilterModel
+	response.Diagnostics.Append(request.State.Get(ctx, &state)...)
+	if response.Diagnostics.HasError() {
+		return
+	}
+
+	if state.CloudProvider.IsNull() || state.CloudProvider.IsUnknown() {
+		response.Diagnostics.AddError("Missing cloud_provider", "cloud_provider is required for lookup")
+		return
+	}
+
+	provider := state.CloudProvider.ValueString()
+	skipCache := true
+
+	params := datadogV2.GetResourceEvaluationFiltersOptionalParameters{
+		CloudProvider: &provider,
+		AccountId:     state.ID.ValueStringPointer(),
+		SkipCache:     &skipCache,
+	}
+	resp, _, err := r.API.GetResourceEvaluationFilters(r.Auth, params)
+	if err != nil {
+		response.Diagnostics.Append(utils.FrameworkErrorDiag(err, "error retrieving CsmResourceEvaluationFilter"))
+		return
+	}
+
+	attributes := resp.Data.GetAttributes()
+	r.UpdateState(ctx, &state, &attributes)
+	response.Diagnostics.Append(response.State.Set(ctx, &state)...)
+}
+
+func (r *CsmResourceEvaluationFilter) Update(ctx context.Context, request resource.UpdateRequest, response *resource.UpdateResponse) {
+	var state ResourceEvaluationFilterModel
+	response.Diagnostics.Append(request.Plan.Get(ctx, &state)...)
+	if response.Diagnostics.HasError() {
+		return
```

Are we swallowing an error here?

I see a few other code paths like this too, but maybe the error is already reported elsewhere?

---

This is the usual usage of the go client, other resources do this exact thing.
Two random files I had open: 

- [file1](https://github.com/DataDog/terraform-provider-datadog/blob/dff9be95b9dd35553ffdabb285c7362f553cd405/datadog/fwprovider/resource_datadog_restriction_policy.go#L156)
- [file2](https://github.com/DataDog/terraform-provider-datadog/blob/358ddf5d45ce26e3697d488d75956e620c5c8b7a/datadog/fwprovider/resource_datadog_team_link.go#L131)

I believe it is because the previous line already adds the context/ error, so all we have left to do if an error is found is return.

---

Ok understood. Thanks for double-checking.